### PR TITLE
Deploy & LoadBalancer controller implementation.

### PR DESCRIPTION
**Pull request by sesh-kebab**
**What does this pull request do?**
Adds implementation at the controller layer for Deploy and LoadBalancer entities.

**How should this be tested?**
Navigate to api and execute `go test ./...`

**Checklist**
- [ x] Unit tests
- [ ] ~Smoke tests (if applicable)~
- [ ] ~System tests (if applicable)~
- [ ] ~Documentation (if applicable)~

closes #
(reference the issue(s) this pull request closes)

### CODE REVIEW

File: api/controllers/deploy.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+			Handlers: fireball.Handlers{
+				"GET":  d.ListDeploys,
+				"POST": d.CreateDeploy,
+			},
+		},
+		{
+			Path: "/deploy/:id",
+			Handlers: fireball.Handlers{
+				"GET":    d.GetDeploy,
+				"DELETE": d.GetDeploy,
```
Comment (zpatrick) [inline api/controllers/deploy.go]: `DELETE` should be `d.DeleteDeploy`

File: api/controllers/deploy_test.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+	req := models.CreateDeployRequest{
+		DeployName: "deploy1",
+		Dockerrun:  nil,
+	}
+
+	DeployModel := models.Deploy{
+		DeployID:   "d1",
+		DeployName: "deploy1",
+		Dockerrun:  nil,
```
Comment (zpatrick) [inline api/controllers/deploy_test.go]: Can we use a non zero-value for `Dockerrun` (`[]byte("content")`)?

File: api/controllers/loadbalancer.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/zpatrick/fireball"
+)
+
+type LoadBalancerController struct {
+	LoadBalancerProvider provider.LoadBalancerProvider
+	JobScheduler         scheduler.JobScheduler
+}
+
+func NewLoadBalancerController(e provider.LoadBalancerProvider, j scheduler.JobScheduler) *LoadBalancerController {
```
Comment (zpatrick) [inline api/controllers/loadbalancer.go]: `e` is an odd name for this param

File: api/provider/loadbalancer.go [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package provider
+
+import "github.com/quintilesims/layer0/common/models"
+
+type LoadBalancerProvider interface {
+	Create(req models.CreateLoadBalancerRequest) (*models.LoadBalancer, error)
+	Read(LoadBalancerID string) (*models.LoadBalancer, error)
```
Comment (zpatrick) [inline api/provider/loadbalancer.go]: [nit] lower-case param names

File: api/controllers/deploy.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/zpatrick/fireball"
+)
+
+type DeployController struct {
+	DeployProvider provider.DeployProvider
+	JobScheduler   scheduler.JobScheduler
+}
+
+func NewDeployController(p provider.DeployProvider) *DeployController {
```
Comment (diemonster) [inline api/controllers/deploy.go]: [nit] `d provider.DeployProvider`

**Pull request merged**